aws/resource_aws_glue_trigger.go

```diff
@@ -365,8 +389,18 @@ func expandGlueActions(l []interface{}) []*glue.Action {
 	for _, mRaw := range l {
 		m := mRaw.(map[string]interface{})
 
-		action := &glue.Action{
-			JobName: aws.String(m["job_name"].(string)),
+		job_name := m["job_name"].(string)
+		crawler_name := m["crawler_name"].(string)
+
+		var action *glue.Action
+		if len(job_name) > 0 {
+			action = &glue.Action{
+				JobName: aws.String(job_name),
+			}
+		} else if len(crawler_name) > 0 {
+			action = &glue.Action{
+				CrawlerName: aws.String(crawler_name),
+			}
```

Comment:
To prevent potential panics if the map does not contain the `job_name` or `crawler_name` keys or they are nil (which may be possible with future resource code/Terraform Plugin SDK changes), Go supports the following syntax to both validate the key exists and is the expected type assertion:

```go
action := &glue.Action{}

if v, ok := m["crawler_name"].(string); ok && v != "" {
  action.CrawlerName = aws.String(v)
}

if v, ok := m["job_name"].(string); ok && v != "" {
  action.JobName = aws.String(v)
}
```

Its also worth nothing that within Terraform resource code, we try not to set our own precedence for configurations (job taking precedence over crawler with if/else in this pull request). We should pass through all values and allow the schema or API validation return invalid configurations instead of (likely) perpetual differences in Terraform if both are declared.